### ISSUE TITLE
Support WebSocket

### ISSUE THREAD

**Issue by nrllh** [wkeeling/selenium-wire#197]
Hi,

WebSocket requests are being sent via `wss ` or `ws` protocol. The output of request.url is always https. Is there anyway to distinguish WebSocket and HTTP requests?

A really great tool, thank a lot for the effort into this project @wkeeling!

**Comment by wkeeling** [comment]
Thanks for flagging this. It does look as though it's an issue and seems to be caused by the proxy that Selenium Wire uses to intercept requests. The proxy seems to override the `wss` with `https`. I'll investigate some more and look at getting a fix in.

In the meantime you should be able to identify websocket requests by looking at the `Upgrade` request header which will be set to `websocket`.
```
for request in driver.requests:
    if request.headers['Upgrade'] == 'websocket':
        print(f'websocket request: {request.url}')
```

**Comment by nrllh** [comment]
@wkeeling thank you very much, this helps too!

**Comment by wkeeling** [comment]
What sort of SSL information are you interested in retrieving?

**Comment by nrllh** [comment]
I'd get the following information:

- Subject
- Serial Number
- Key
- Signature algorithm
- Trusted
- Issuer

I think there are some possibilities by creating an extension, but I run selenium for different browsers, so it's not the best way for me. I also couldn't see any function for extracting such info. I'd be for my research very interesting.

**Comment by wkeeling** [comment]
Thanks. This may be possible, would need some investigation. I guess ideally you'd be looking for a response attribute, e.g. `response.tls` from which you could access the above attributes.

**Comment by nrllh** [comment]
I run:

  ```
  for r in driver.requests:   
        if r.response:
            if r.response.tls:
                print(r.response.tls)
```
and get:

> AttributeError: 'Response' object has no attribute 'tls'

Am I doing something wrong?

**Comment by wkeeling** [comment]
Ah the `response.tls` isn't actually implemented, I was proposing it as an api. Perhaps `response.tlsinfo` may be clearer. Anyway, I'll investigate how we can capture the info and will post back here once something is implemented. Thanks.

**Comment by nrllh** [comment]
Thanks, but also `tlsinfo` seems not to exist. I run:

```

    for r in input_driver.requests:   
        if r.response: 
            try:
                if r.response.tlsinfo:
                    print(r.response.tlsinfo)
                    exit()
            except:
                print('I"m not tls')
```

**Comment by wkeeling** [comment]
Websockets are now fully supported from v4.1.0 onwards. Have raised #234 to track capturing SSL meta-data.

**Comment by wkeeling** [comment]
@nrllh SSL certificate information is now available from v4.2.1 using `request.cert`. 
Tracked by https://github.com/wkeeling/selenium-wire/issues/234
